client/src/components/TaskCard/TaskCard.js

```diff
@@ -77,6 +78,7 @@ const TaskCard = ({
       )}
       {showReflection ? (
         <TaskReflectionModal
+          readOnly={finished}
```

Comment:
I would rename `finished` to `taskCompleted` or something similar

Reply:
Renamed.